dist2src/core.py

```diff
@@ -0,0 +1,170 @@
+            "set -e; "
+            "shopt -s dotglob nullglob && "  # so that * would match dotfiles as well
+            "cd BUILD/kernel-4.18.0-*.el8/ && "
+            "mv ./linux-4.18.0-*.el8.x86_64/* . && "
+            "rmdir ./linux-4.18.0-*.el8.x86_64 && "
+            "git add . && "
+            "git commit --amend --no-edit"
+            # the patch is already applied in %prep
+            # "git apply ../../SOURCES/debrand-single-cpu.patch &&"
+            # f"git commit -a -m '{KERNEL_DEBRAND_PATCH_MESSAGE}'"
+        )
+    },
+    "pacemaker": {INCLUDE_SOURCES: ["nagios-agents-metadata-*.tar.gz"]},
+}
+
+
+def get_hook(package_name: str, hook_name: str) -> Optional[str]:
+    """ get a hook's command for particular source-git repo """
+    return HOOKS.get(package_name, {}).get(hook_name, None)
+
+
+class GitRepo:
+    """
+    a wrapper on top of git.Repo for our convenience
+    """
+    def __init__(self, repo_path: Optional[Path]):
+        self.repo = None if not repo_path else git.Repo(repo_path)
+
+    def checkout(self, branch, orphan=False):
+
+    def commit(self, message: str):
+
+    def commit_all(self, message: str):
+        if self.repo.is_dirty():
+            self.stage()
+            self.commit(message=message)
+        else:
+            logger.info("The repo is not dirty, nothing to commit.")
+
+    def fetch_branch(source_dir, dest_dir, dest_branch):
+
+    def cherry_pick_base(gitdir, from_branch, to_branch):
+
+    def stage(gitdir, add=None, exclude=None):
+
+    def create_tag(tag, path, branch):
+
+
+
+class Dist2Src:
+    """
+    A convertor for dist-git rpm repos into a source-git variant.
+    """
+
+    def __init__(self, dist_git_path: Optional[Path], source_git_path: Optional[Path], log_level: int = 1):
+        """
+        both dist_git_path and source_git_path are optional because not all operations require both
+
+        @param dist_git_path: path to the dist-git repo we want to convert
+        @param source_git_path: path to a source-git repo (doesn't need to exist)
+                                where the conversion output will land
+        @param log_level: int, 0 minimal output, 1 verbose, 2 debug
+        """
+        self.dist_git_path = dist_git_path
+        self.dist_git = GitRepo(dist_git_path)
+        self.source_git_path = source_git_path
+        self.source_git = GitRepo(source_git_path)
+        self.log_level = log_level
+
+    @property
+    def package_name(self):
+        return self.dist_git_path.name
+
+    def get_archive(self, get_sources_script_path: str = "get_sources.sh"):
+        """
+        Fetch archive using get_sources.sh script in the dist-git repo.
+        """
+        command = sh.Command(get_sources_script_path)
+
+        with sh.pushd(self.dist_git):
+            logger.info(f"Running command {get_sources_script_path} in {os.getcwd()}")
+            stdout = command()
+
+        logger.debug(f"output = {stdout}")
+
+    def run_prep(self):
```

Comment:
no strong preference on my side, can do it

Reply:
Thanks. It would be nicely readable when we want to combine various phases in the CLI.

e.g.:

```python
dist2src.run_prep().commit_all().add_config().commit()
```